adapters/nextmillennium/nextmillennium.go

```diff
@@ -0,0 +1,162 @@
+	for _, imp := range imps {
+		impExt, err := getImpressionExt(&imp)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		if err := validateImpression(impExt); err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		resImps = append(resImps, impExt)
+	}
+
+	return
+}
+
+func validateImpression(impExt *openrtb_ext.ImpExtNextMillennium) error {
+	if impExt.PlacementID == "" {
+		return &errortypes.BadInput{Message: "No valid placement provided"}
+	}
+	return nil
+}
+
+func getImpressionExt(imp *openrtb2.Imp) (*openrtb_ext.ImpExtNextMillennium, error) {
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: err.Error(),
+		}
+	}
+	var NextMillenniumExt openrtb_ext.ImpExtNextMillennium
+	if err := json.Unmarshal(bidderExt.Bidder, &NextMillenniumExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: err.Error(),
+		}
+	}
+
+	return &NextMillenniumExt, nil
+}
+
+func (adapter *adapter) buildAdapterRequest(prebidBidRequest *openrtb2.BidRequest, params *openrtb_ext.ImpExtNextMillennium, imps []openrtb2.Imp) (*adapters.RequestData, error) {
+	newBidRequest := createBidRequest(prebidBidRequest, params, imps)
+
+	reqJSON, err := json.Marshal(newBidRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+
+	return &adapters.RequestData{
+		Method:  "POST",
+		Uri:     adapter.endpoint,
+		Body:    reqJSON,
+		Headers: headers}, nil
+}
+
+func createBidRequest(prebidBidRequest *openrtb2.BidRequest, params *openrtb_ext.ImpExtNextMillennium, imps []openrtb2.Imp) *NextMillenniumBidRequest {
+	bidRequest := NextMillenniumBidRequest{
+		ID:   prebidBidRequest.ID,
+		Test: uint8(prebidBidRequest.Test),
+	}
+	bidRequest.Ext.Prebid.StoredRequest.ID = params.PlacementID
+
+	return &bidRequest
+}
+
+//MakeBids translates NextMillennium bid response to prebid-server specific format
+func (adapter *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	var msg = ""
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+	if response.StatusCode != http.StatusOK {
+		msg = fmt.Sprintf("Unexpected http status code: %d", response.StatusCode)
+		return nil, []error{&errortypes.BadServerResponse{Message: msg}}
+
+	}
+	var bidResp openrtb2.BidResponse
+	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
+		msg = fmt.Sprintf("Bad server response: %d", err)
+		return nil, []error{&errortypes.BadServerResponse{Message: msg}}
+	}
+	if len(bidResp.SeatBid) != 1 {
+		var msg = fmt.Sprintf("Invalid SeatBids count: %d", len(bidResp.SeatBid))
+		return nil, []error{&errortypes.BadServerResponse{Message: msg}}
+	}
```

Comment:
Please add json test coverage for the `len(bidResp.SeatBid) != 1` condition.

Reply:
@SyntaxNode Done!